### PR TITLE
Remove post migrate handler with a default instance creation via Migration

### PR DESCRIPTION
**Pull request by VaZark**
This functionally works when running migrations and skips the default theme creation issue on app installation.

- [ ] Fix tests

### CODE REVIEW

File: admin_interface/migrations/0001_initial.py [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
 
 class Migration(migrations.Migration):
+    def create_default_theme(apps, schema_editor):
```
Comment (merwok) [inline admin_interface/migrations/0001_initial.py:10]: This should be a function, not a method

File: admin_interface/migrations/0001_initial.py [hunk 2]
```diff
@@ -225,4 +228,5 @@ class Migration(migrations.Migration):
                 "verbose_name_plural": "Themes",
             },
         ),
+        migrations.RunPython(create_default_theme),
```
Comment (merwok) [inline admin_interface/migrations/0001_initial.py:231]:
```suggestion
        migrations.RunPython(create_default_theme, reverse=migrations.RunPython.noop),
```
to allow reversing the migration

File: admin_interface/migrations/0001_initial.py (hunk 1 above)
Comment (VaZark) [reply admin_interface/migrations/0001_initial.py]: _Outdated_ Replaced it with raw sql

File: admin_interface/migrations/0001_initial.py (hunk 2 above)
Comment (VaZark) [reply admin_interface/migrations/0001_initial.py]: Added it to the updated version

**Comment by VaZark** [comment]
even though not the most elegant method, I chose to go with raw sql. 

Using `apps.get_model("admin_interface", "Theme")` throws errors due to inconsistency between the the table schema at the moment of migration (after 0001_initial) and the current shape of the model in the code. 

I'm open to any alternate ideas for inserting data during migration without jumping through signals tho

**Comment by VaZark** [comment]
note : Tests are currently broken on sqlite for django < 2.0 .. looks some kind of error parsing multiline SQL statements (only for sqlite)?

**Comment by merwok** [comment]
I don’t think it’s correct to modify a migration that’s already deployed.

Why not add a new migration, and use apps.get_model without issue?

**Comment by VaZark** [comment]
The app_model can change again in the future. Then it will be inconsistent again :(

I updated the raw SQL first migration as that's the least risk for inconsistency. Adding it later risks accidental data-overwriting shenanigans

**Comment by merwok** [comment]
> The app_model can change again in the future. Then it will be inconsistent again

No it will not?  People run migrations up to the one that creates a default Theme, then new migrations will add fields with default value, so nothing breaks.

**Comment by VaZark** [comment]
The result of the app_model returns the fields configured in code. There have been some field renaming in the past migrations. 

So if there are any new renames or removals in the models, that breaks the result from app_model vis-à-vis the migration state. Insert stmt will be run with columns and values not in the table as well.

So just by adding the raw sql in the initial migration, i can ensure that new migration is consistent and old ones are never touched.

**Comment by VaZark** [comment]
I’m open to any alternative ideas that account for rename and field removal. 👍

**Comment by merwok** [comment]
Editing deployed migrations is a django no-go.  I am not going to migrate my servers to 0001 then up again and lose my data.

I am sorry but I do not understand your point about using apps.get_model.  I use if quite often to populate data.
Migrations 0001 to for example 0006 bring the table to a specific schema, then the class retrieved with apps.get_model in migration 0007 conforms to that schema and can be used to create records, then if I add or change fields in migrations 0008 and 0009 they will be added or changed in the existing records.  I never edit the code in 0007 to change what’s there, it’s always running with the schema of that step.

**Comment by VaZark** [comment]
Hmm.. maybe I’m missing something up.

This is the error message i see when i use app_model in the initial migration. 

```
2022-12-01T12:59:13.0909724Z     return self.cursor.execute(sql, params)
2022-12-01T12:59:13.0910132Z django.db.utils.ProgrammingError: column "css_header_title_color" of relation "admin_interface_theme" does not exist
2022-12-01T12:59:13.0910560Z LINE 1: ...", "logo_visible", "css_header_background_color", "css_heade...
```

You can checkout the commit results [here](https://github.com/fabiocaccamo/django-admin-interface/pull/219/commits/329dafa5c83095430bae62ad1f51a695eae6019f)

I’ll take a another look tmrw

**Comment by VaZark** [comment]
Added new commit with two migrations (one with app_model and second with field rename) that show why tests break if we go with the app_model approach.

The issue has never been the logic itself. I'm completely on board with that. It's more about the tests, they don't even start :(

**Comment by fabiocaccamo** [comment]
I think that here we are really overcomplicating the problem, I will drop the `post_migrate` handler in the next release and let's see the next open issue :).

**Pull request closed without merging**